src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,359 @@
+import { Vector } from 'ol/layer';
+import { Circle, Stroke, Style, Fill } from 'ol/style';
+import { Vector as VectorSource } from 'ol/source';
+import { Feature } from 'ol';
+import { LineString, Point } from 'ol/geom';
+import { Modify } from 'ol/interaction';
+import { unByKey } from 'ol/Observable';
+import EventType from 'ol/events/EventType';
+import { click } from 'ol/events/condition';
+
+import { fromLonLat, toLonLat } from 'ol/proj';
+import lineSlice from '@turf/line-slice';
+import { getDistance } from 'ol/sphere';
+import {
+  lineString as createTurfLine,
+  point as createTurfPoint,
+} from '@turf/helpers';
+import RoutingAPI from '../../api/routing/RoutingAPI';
+import Control from '../../common/controls/Control';
+import Layer from '../layers/Layer';
+
+const defaultPointStyle = new Style({
+  image: new Circle({
+    radius: 6,
+    fill: new Fill({
+      color: [255, 0, 0, 1],
+    }),
+  }),
+});
+
+/**
+ * Display layer's copyrights.
+ *
+ * @example
+ * import { Map, RoutingControl } from 'mobility-toolbox-js/ol';
+ *
+ * const map = new Map({
+ *   target: 'map',
+ *   controls: [
+ *     new RoutingControl()
+ *   ]
+ * });
+ *
+ *
+ * @see <a href="/example/ol-routing">Openlayers routing example</a>
+ *
+ * @extends {Control}
+ * @implements {RoutingInterface}
+ */
+class RoutingControl extends Control {
+  constructor(options) {
+    const opts = {
+      ...options,
+    };
+
+    if (!opts.element) {
+      /**
+       * Define a default element.
+       */
+      opts.element = document.createElement('button');
+      opts.element.id = 'ol-routing';
+      opts.element.innerHTML = 'Toggle Route Control';
+      Object.assign(opts.element.style, {
+        position: 'absolute',
+        right: '10px',
+        top: '10px',
+      });
+
+      opts.element.addEventListener(
+        EventType.CLICK,
+        () => this.setDrawEnabled(!this.active, true),
+        false,
+      );
+    }
+    super(opts);
+
+    this.abortController = new AbortController();
+
+    this.apiKey = opts.apiKey;
+
+    this.api = new RoutingAPI({
+      apiKey: this.apiKey,
+    });
+
+    this.mot = 'bus';
+
+    this.active = opts.active || true;
+
+    this.routingLayer =
+      opts.layer ||
+      new Layer({
+        name: 'routing-layer',
+        olLayer: new Vector({
+          source: new VectorSource(),
+          style: [
+            new Style({
+              stroke: new Stroke({
+                color: [255, 0, 0, 1],
+                width: 5,
+              }),
+            }),
+          ],
+        }),
+      });
+
+    this.viaPoints = [];
+
+    this.setDrawEnabled(this.active);
+  }
+
+  setDrawEnabled(enabled, reset) {
+    this.active = enabled;
+
+    // Clear viaPoints and source
+    if (reset) {
+      this.viaPoints = [];
+      this.routingLayer.olLayer.getSource().clear();
+    }
+
+    if (this.map) {
+      if (!this.active) {
```

Comment:
use activate/deactivate function see others controls for an example

Reply:
done